rust/agama-dbus-server/src/questions.rs

```diff
@@ -0,0 +1,215 @@
+use std::collections::HashMap;
+
+use crate::error::Error;
+use agama_lib::connection_to;
+use anyhow::Context;
+use zbus::{dbus_interface, zvariant::ObjectPath, fdo::ObjectManager, Connection};
+
+#[derive(Clone,Debug)]
+pub struct GenericQuestion {
+    id: u32,
+    text: String,
+    options: Vec<String>,
+    default_option: String,
+    answer: String,
+}
+
+impl GenericQuestion {
+    pub fn new(id: u32, text: String, options: Vec<String>, default_option: String) -> Self {
+        Self {
+            id,
+            text,
+            options,
+            default_option,
+            answer: String::from("")
+        }
+    }
+
+    pub fn object_path(&self) -> String {
+        format!("/org/opensuse/Agama/Questions1/{}", self.id)
+    }
+}
+
+#[dbus_interface(name = "org.opensuse.Agama.Questions1.Generic")]
+impl GenericQuestion {
+    #[dbus_interface(property)]
+    pub fn id(&self) -> u32 {
+        self.id
+    }
+
+    #[dbus_interface(property)]
+    pub fn text(&self) -> &str {
+        self.text.as_str()
+    }
+
+    #[dbus_interface(property)]
+    pub fn options(&self) -> Vec<String> {
+        self.options.to_owned()
+    }
+
+    #[dbus_interface(property)]
+    pub fn default_option(&self) -> &str {
+        self.default_option.as_str()
+    }
+
+    #[dbus_interface(property)]
+    pub fn answer(&self) -> &str {
+        &self.answer
+    }
+
+    #[dbus_interface(property)]
+    pub fn set_answer(&mut self, value: &str) -> Result<(), zbus::fdo::Error> {
+        // TODO verify if answer exists in options or if it is valid in other way
+        self.answer = value.to_string();
+
+        Ok(())
+    }
+}
+
+#[derive(Clone,Debug)]
+pub struct LuksQuestion {
+    password: String,
+    attempt: u8,
+    generic_question: GenericQuestion,
+}
+
+impl LuksQuestion {
+    fn device_info(device: &str, label: &str, size: &str) -> String{
+        let mut result = device.to_string();
+        if !label.is_empty() {
+            result = format!("{} {}", result, label);
+        }
+
+        if !size.is_empty() {
+            result = format!("{} ({})", result, size);
+        }
+
+        return result.to_string();
```

Comment:
In Ruby we omit the `return` word whenever we can. Why not doing the same here?

```suggestion
        result.to_string()
```